wp-dependency-installer.php

```diff
@@ -616,22 +652,34 @@ function ( $plugin_file ) {
 		 * @return mixed
 		 */
 		public function unset_action_links( $actions ) {
-			if ( isset( $actions['edit'] ) ) {
-				unset( $actions['edit'] );
-			}
 			if ( isset( $actions['delete'] ) ) {
 				unset( $actions['delete'] );
 			}
-			if ( isset( $actions['deactivate'] ) ) {
-				unset( $actions['deactivate'] );
-			}
 
 			/* translators: %s: opening and closing span tags */
 			$actions = array_merge( [ 'required-plugin' => sprintf( esc_html__( '%1$sRequired Plugin%2$s' ), '<span class="network_active" style="font-variant-caps: small-caps;">', '</span>' ) ], $actions );
 
 			return $actions;
 		}
 
+		/**
+		 * Prevent admin ‘?action=deactive’ for required plugins.
+		 *
+		 * @return void
+		 */
+		public function before_deactivate_plugin() {
+			global $current_screen;
+			if (
+				isset( $_REQUEST['plugin'] ) &&
+				isset( $current_screen ) &&
+				in_array( $current_screen->id, [ 'plugins' ], true ) &&
+				$this->is_required( $_REQUEST['plugin'] )
+			) {
```

Comment:
I'm not sure that all these conditions are necessary, but I wanted to be on the safe side ...

Reply:
now it should be a little easier to understand